File: 2024/1/sort.cc
#include <algorithm>
#include <cmath>
#include <cstdio>
#include <unordered_map>
#include <vector>

int main() {
  std::vector<int> list1, list2;
  std::unordered_map<int, int> counter2;
  int a, b;
  while (scanf("%d %d", &a, &b) == 2) {
    list1.push_back(a);
    list2.push_back(b);
    counter2[b]++;
  }

  std::sort(list1.begin(), list1.end());
  std::sort(list2.begin(), list2.end());

  long long solution1 = 0, solution2 = 0;
  for (size_t i = 0; i < list1.size(); ++i) {
    solution1 += std::abs(list1[i] - list2[i]);
    solution2 += 1LL * list1[i] * counter2[list1[i]];
  }

  printf("%lld\n%lld\n", solution1, solution2);
  return 0;
}

File: 2024/1/sort.py
import sys
import collections

if __name__ == "__main__":
    # Read input line by line from standard input
    numbers = [[int(t) for t in line.split()] for line in sys.stdin]
    # Build the two sorted lists from each column
    list1, list2 = [sorted([r[c] for r in numbers]) for c in range(2)]

    # First solution is the sum of the pairwise absolute difference
    solution1 = sum(abs(n1 - n2) for n1, n2 in zip(list1, list2))
    print(solution1)

    # Count the number of times each element appears in the second list
    counter2 = collections.Counter(list2)
    # Second solution is the sum of the product of the number in the first list
    # by its appearance in the second list
    solution2 = sum(n1 * counter2[n1] for n1 in list1)
    print(solution2)
